### ISSUE TITLE
engine_install_url defaults to null

### ISSUE THREAD

**Issue by DJAyth** [rancher/terraform-provider-rancher2#79]
As discussed on the Rancher Slack channel.

I've been working on a deployment from the ground up of a Rancher instance, and ran into what could be described as a bug.

The documentation for the rancher2_node_template resource lists engine_install_url, as an optional parameter. As such I omitted it from my deployment, however, when I attempted to create a cluster with rancher2_cluster it consistently failed with an error that it was unable to install Docker.

Here is my node template portion of my tf file(a few things sanitized):

```
resource "rancher2_node_template" "node-template-sto3" {
  name                  = "node-template-sto3"
  openstack_config {
    auth_url            = "https://XXX.XXXXX.XXXX:5000/v3"
    availability_zone   = "av_zone"
    region              = "region"
    username            = "domain@domain.com"
    domain_name         = "Default"
    flavor_id           = "83d8b44a-26a0-4f02-a981-079446926445"
    image_id            = "9d7efc49-d5f4-4fb4-8a42-a7c082404a32"
    net_id              = "fe434879-33a2-4aaf-bb2c-60667a592156"
    password            = var.openstack_password
    ssh_user            = "ubuntu"
    tenant_name         = "tenant_name"
    sec_groups          = "86bd43c4-6afc-469f-b59d-152cfc2613aa,55e080ae-5e74-4b1f-855e-fd97f0ed1048,86b640da-b301-425c-84e9-e31bfeb2eec6"
   }
  depends_on            = [ "rancher2_node_driver.openstack"]
}
```

Fairly quickly tracked it down to the missing engine_install_url, in the UI the script for installing docker was empty, so I entered in the script and it went forward.

When you create a node template with the UI the value is pre-populated for you, as such I believe it should be either a required parameter, or it should have some type of a default. Rancher somewhere does have a set of defaults, as it populates them in the UI. Possibly that could be pulled?

**Comment by rawmind0** [comment]
Fix issue on PR #81

**Comment by rawmind0** [comment]
Merged PR. Please, reopen issue if needed.
